frontend/pages/SoftwarePage/SoftwareAddPage/SoftwareAddPage.tsx

```diff
@@ -71,12 +72,28 @@ const SoftwareAddPage = ({
     [location, router]
   );
 
+  // Quick exit if no team_id param. This page must have a team id to function
+  // correctly. We redirect to the same page with the "No team" context if it
+  // is not provieded.
+  if (!location.query.team_id) {
```

Comment:
Was this missed in the specs? It looks like the URL routes in the Figma mock ups don't contemplate this? I guess it was a modal before so we didn't need to worry about it. Now that it is its own page we do need to handle the cases where we don't know the team id (e.g., direct navigation to the page). But maybe we should revisit the UI design to give a better indication of the team where the software will be added? We'd normally pair the team_id param with the teams dropdown, but in this case it is competing for real estate with the back link.

Reply:
This was not covered in the specs but @marko-lisica and I had a conversation about this and it was decided this is how we handle this now and make an improvement on it in the future. I'll add a dev note in the figma so its documented.